### PR TITLE
Fix misc aarch64 issues

### PR DESCRIPTION
**Pull request by Keno**
Just a collection of misc aarch64 issues I saw in the tests, plus two fixes for prior commits that already got merged, but had issues.

### CODE REVIEW

File: src/Task.cc [hunk 1]
```diff
@@ -1920,6 +1920,19 @@ void Task::canonicalize_regs(SupportedArch syscall_arch) {
   registers_dirty = true;
 }
 
+bool Task::aarch64_read_tls_register(uintptr_t *result) {
+  struct iovec vec = { result,
+                    sizeof(*result) };
```
Comment (khuey) [inline src/Task.cc:1925]: Can you put this all on one line or do something else sensible format wise?

File: src/Task.cc [hunk 2]
```diff
@@ -1920,6 +1920,19 @@ void Task::canonicalize_regs(SupportedArch syscall_arch) {
...
+  return ptrace_if_alive(PTRACE_GETREGSET, NT_ARM_TLS, &vec);
+}
+
+void Task::set_aarch64_tls_register(uintptr_t val) {
+  struct iovec vec = { &val,
+                       sizeof(val) };
```
Comment (khuey) [inline src/Task.cc:1931]: here too

File: src/Registers.cc [hunk 3]
```diff
@@ -687,7 +687,7 @@ bool Registers::aarch64_singlestep_flag() {
 void Registers::set_aarch64_singlestep_flag() {
   switch (arch()) {
     case aarch64:
-      return set_pstate(pstate() & AARCH64_DBG_SPSR_SS);
+      return set_pstate(pstate() | AARCH64_DBG_SPSR_SS);
```
Comment (khuey) [inline src/Registers.cc:690]: oof

**Comment by Keno** [comment]
Fixed the formatting nit. Also renamed `aarch64_read_tls_register` to `read_aarch64_tls_register`, for consistency.

**Pull request merged**